### PR TITLE
[AGD-1882] Support storing user defined graph documentation

### PR DESCRIPTION
**Pull request by SHKnudsen**
### Purpose

This pull request addresses JIRA task [AGD-1882](https://jira.autodesk.com/browse/AGD-1882).

Example of `.dyn` file storing a Base64 thumbnail:
```json
{
  "Uuid": "35aa6c06-f083-4e4f-9817-ed5d234c4ca2",
  "IsCustomNode": false,
  "Description": null,
  "Name": "thumbnailTest3",
  "ElementResolver": {
    "ResolutionMap": {}
  },
  "Inputs": [],
  "Outputs": [],
  "Nodes": [],
  "Connectors": [],
  "Dependencies": [],
  "NodeLibraryDependencies": [],
  "ExtensionWorkspaceData": [],
  "Thumbnail": "iVBORw0KGgoAAAANSUhEUgAAAIQAAACmCAYAAAAbICeaAAAVRElEQVR42uzc+VNUVxYHcH/PD/ktP6Zq/oBUhRgV2ZFFZFFkFUVAaZYGWZRNaLZmb3aQHUEiuOLuuIvGNSQxY5aZJCaxyjFTZZxJZmI5mZhJMmfeIdGjPqW7733Lfdi36ltlReVVeT7ce965pOcpsT72D35ZSthH/sGN1/2DTlzzC7rxgV/Qj0/nT/5BJ6XfH5D+rPlycJz7PNeaWwsRSAXeJQF48K5f4K/TfoHgaN7zC/z5SkD49ydC1h5rjaqpbFnVs1zKa/Ncy5AQVkvf7Z9JRf2FisyXXUHxYFm++duyldab5VG1B6wxNhcSA0B4FY8EgqBsLvkvhf6lKZAfUfwoRSvKHyGpiWm2SUASEImUl+a5lr67gnQ03MXCqZ39gTFQHF7wOIxnIqmIqjtVFd3Q7kKi8ZKaxT7aFbTJySURYA3LIQh2Urii7N7myKrblqiaKRcSUTHwHyHPQuFCYmwM/CjKwjc+XWwXEj16BsKgb84uCYeSCFlP4UKiIYbXqIEUI4eCYmBTRNHTxXQh0WLhq6U4GCjDy9Y9C4ULidpHBU0dxcrZgAjoCMt6FgoXErUWTiDFw0DZGZL4PBQuJGrcTSixO3xsyoC/jW2HO5P74Z+Xrsz8+mZTM7wfHqnILtEXlvE8FC4kCvcOu3iKdaO8Eh7c+QZmWYiEG8bW0NTZULiQKHWFjbeWrEX6+4lT4OBCNLiLMIPYF7wKQXCiEB9JdXTj4O9I3KW8ItxxwY+B1i/378P1VWt4jo05h8KcWAfJ+X2wqnoCopsPQGTHEVg2+jYs23bxl8jBc/dies/eXD566WjYxLulYTveC5PysmogPvQPbmI9JljXvesfsu4SCMHgKAgBAggbPIPFdziIJPStyw9Cx9/ZJcFYLczsAbd/nvVpfgETiPGQJEOjSF9nw10Ai8ufbRd+Dd1+9TNFYeCPvbG8TXAuPG54QBgORU5MOcTW7+EAYBfGpxIML24Q+DOPzhblVk8fuwRqMJlAHAyOIwhyFEJiMKW30dGgYkLHLv1X6jMaNQeB8wUFloIgKE3hOUJhWFs8jMXSMBf+J/UXx7HxdIEQCwU2jVQo7VFcRRTig6DXTx4QwqNYbRnTCQOhCBufvuI0imsMIPANgXPhaJsHhNAo1mV36wdBvlMMOLdD+Ph9w1IY/A7nWHjHwQNCWBTZCVa1G0in5xZhE9NZjoPw9rnFUBg8NnjfMHhAiIlixWaIajssCAaKNMi6Kx0drzoEYtrD/QZrcX748ivWoRQvCCFRJBfQG4VokfqJrQ6BmFo0nxUE3l4iCpajgheEcCg2JNRDRP9p8TDQLvGTtEvY/7/hTi54/cYFHz/gQYFX23YWwqGbTn4QQqHIja6AxDJlXjFT9k5D28XPYeL6rZkMTn8F2YeuKbNLTEz3OwTiqIc7b6HwBhMnmHhxhcXHphN/jSNqvAijP6sMCGFQ5EWWgdnUBdGtR7iKVXL8Q/j4zr/geeub+z8iFN5d4gG+htoFsXOhG1z1DcB/cLFDIIRBkbWuHdLNfVyFOviXr8HRhWhiJi4z33ngRZhdEBMLXofjnh4vCAhKZXg+3ytmYhNkpm3B40ILDNSHfXefGQVemzsEAneJy75LjAJCdxQ5cVbEMJNY20HmY4JxISQ2ENuv3HEIBOag+0KjgNAVRV6UBfuGRyBWdh5jKg72BRwLG1CmG1GaSdgBgTnp6WkEEDqgoOGTeX0HQngYJgz45kBLw11i7OLPEggvB0BQ8DVUcBC6ochKbkUE3CDwlZJ5UYPJdL8RO3hxqTMgsJ8gFMYHwY+Chk8EgcLeTPIvpmdLO8RmuyAMgYJAaI4iN7aKEPCDoJmDMUAQivPePkYGwY+Chk+EQIwjA5tSpmcHT17LZwKB2SHlrJcHvOMX8OKAIBTYROLwiQAoBAKnjrzr6l//wfRsnwPvljCCIBQnPRbCJR/fFwQEpTixDouuOAgcLHEu5lG255H3BzlAEIoTHgtgyssdLvr44I4x50F0xBRCjckGm9I6ZgOBt5zaNpZ0XDDF7eB5HhCU3VJPcdpzIZz5Pee9PWZwXPX1n3MguiNzEMOj5KTJegj2wRTtEqzDKZxyMj0zaOQ0uE2eHuUGIUchzzmp17jg7YVHCyIxLIjeCDPUrm98AkS1qRlRKAiCBlT//ulnlY8Kiu/wUXDbcfSiXRDcKGZB8ra358xOcoWQCA2iMclKGOQolANBOwW9hto5JmhnYMvCgd3gtvv4DWYQ/CjkmfJaLEMiCoiWhDJC4CCK+Pp9HEWiYLHPfHnn8WMEdw98m6BdgSPBI2fBrW87AwheFPxIdAHRGb2Jiu8EioSavfgPLnw8BvcjCDwybvKC4EfBj4RAqNNEUt/gBArxQdDuML9/4jcQo5Pful345BVuEPwo+DPo7QMWv1CoDYiClqVrFGsiG5JrqeDOoUAQxtkdCMRrDCDEQ7HFYxFkL3Z/IgVe/lxImteUU6EZUKSWbBMaQ8DWUwhBUxCEQksQFGYkbXHFjBgolrxuiBw+KySGkNFz8Gb/zidBDO38QQsQhEJ7EExIulbmc0CgVOR2QXbbXiFRuA9MIgJZJBDLGUAIhoJA8MfTE0yxaZCVkAt5iYVQlGQByzorM4ic1j3CofAcPEgA5CASNABBmVz0htAg0lYmgyk+81lhQYIgHqEIGzmvOwbvoSNYeBFAUPYjCgFBZIRGPw8DIxICgUnv2K8bipDR8+Az+Ed4o298dhBHL6VqC4JQCAUiMzDUPgAGJNktO3VHETwyBd6DR2F+3w4ZAFkOTLVrD4JQCAEiy9cf+wYspuJJrR+G1KZRSLONQbrtLchsngBT+6SE4pzqEJZKz/AfOgFeA0dgQf9uuxgweAWuDwhCoS8IT0/qG9QAUd6FIGRJbNkOfgNHwW/oOAQMn8LvYtzWFYeAce/fRwUXGgSh0A1E+vIELJzGIAiFT/9hLNrD4DnPgAQRTEl/57T094898fU8BuiNwqFMHNmlLwhCoTmIzJBILJrGIOyjoBASXwnJQygY3AHwvxEAeTwHDktN5IRTIPDGU38QhEIzEGb/QDUhEIjSNgLAgIIjNIk0JAhCoTqILG8faiLVTn4tFl1zFAv792CBDQ6CUKgJgppI7UBoimJxP91gOp2xA3fEAkEoVAGRER6nHQYCoT4KaiJp+MQQuvEUBgTlkPt85UDQ8EnbmMsdB0EoWDBgE0nDp7kFgnJk8ZvsIPiHT/xJLXYCAyWmfadzIGj4NIdBEApWEDR8il6vPQYCoTqKRf2T/BjogstdXBCEgg0EDZ9EBMGPgoZPSoJIEBsEoXAOBA2f9MvaPCysKiho+DT+AoIgFA6DMC9Zqh8EChZVLRTURCqZ0+8nGQMEobADgoZPgoLgR0HDJ+Wz91QJAwhBUBAIvhtM8UEQCrrBZAjTjSeBMAgKAkHDJ3FAVPUoiiK2bTv2DS4Qs6BAEKzDJy1vPPnTOALm6haIbexSDQR+LIAxQRAKBIHDJyzAnAaRaW2HrMoGjGoo8GMBDAqC0uLhjsOnOQ0iva7nIQaVUNCNp+FB5ITHw9q4DDFBFDRwY0hrGCAITyW4bcAF4vH0Lw0HU0wq+EabxENBN57MMTUOQ1ZVEyFQG8X4oduGBTHi6wetsetnQHhFmwjF3AGBTSQVX20UdMH1B8OBGPNcDO3RyY+DkKEwOoiMmg4quiYo6MbTUCDGF82HrhUJiEEGQjgUqcWMTWQvFZsPxdwH0btsBUKQg5CjMCQIU8MQ9Q2aoaCPBTAUiKGAYMIgA8GAQhwQNHyqslGB9UBBHwtAIATtGwiCHIRYKAgE6/BJPxR0BU4gBOwbsIm0A0JIFI43kbXdVFBVURgfBDWRDCB0RsEwfNIZxf6zGWKCoOETAWAAoTcK/uETJbe6Fgqs1WCpK5WlpLYcf48DBd14Cgti2D8Ai84JghIYkwbrtQZRP2zvBnNWBFjoqoZiaLblQmvzBofS0LQRKupLIL/a+ujreHWPGBgEDZ8YQAiCgi64nB0+YSHtIXAKB6JKr25CFGwgBBo+KQ2CUOgMgoZPdByU1lqgyZbHDsAOjLxaK6Kw+7EAwoHoCYvCYqsFglBoD4KGT3QkgLWhEIumScoaSsGne9jOjacoIGj4xABCTBSpxc2yviG/po7jSOBPja0A/LYMiQ9i1Nsbi8wCQlAUdMGVZhuB3NY+qG3ejEXRPQ0tebCkZ0BcEG8tXohNJCsIYVFkF9dAYXs3WDpboa6tnIoiIgq64HpJbxDURLKBEApFZvIG2JhXAGWVZWCx1c1gqG6voWIIhmJR7zbZjad+IOgGkweE7ijSVpsh15wPJaUlYLFaKI1WqOxogJZm6hdES1FbmSAgaPjEC0I3FNmmXCgsLCIAT6W8rgJsLZsExUBJ7mz8DcTWPff0AUHDJ5FAEApHjwSrZdY0VJkFx0BHh1fvMH0sgHYgaPjUsTIRiyoQCMqymDTZkZCTkUdHggOpKc+C1rLVgmOgZHVU042npiBo+CQgCEpYbPrMkbBpY6GDCCiVVYXQWhQDrcVxqhVwaLBS6a+Ju4T2IAaClmExhQXhtyYTVhQVQkJTBWQ2VzqNoby6FFoQAoLAKAjgg2vn4d7338HT68svPoITx8eV6CXoYwEIhOrDJ+FA+MSnQ0hWHsTXWWBtR/XjcRpF0+YkhKAoiHNT++DBg/+AvXX37tewfayJ+TnW1gJw23emVn0QNHwSCkRweg6sLC2WIWBFUV+WhggUBfHnT6bBiYVwuFBETB4YZAXBf4PJD4LnSJAVngeFtWIDApCnIZO5OHgMsCxEsaWriOmZeds6J9QFQcMnNUDwHgl8KKiJpL7h6dSnMxUGC4qFZVy4szA919ZZeEw9EDR80gVE4PpsO0cCDwpqIm0lawgANwjqGzgX0y5R01N6Sx0Q9OPzGoGgIyFi4ybZkaAWisbSFCy84iBu3/6CG8Shg0NOP7ejLfdH5UHQ8EkDEHQkxFaXMhSYGQUOn2QAZKlOYQKhxLp65bg4ILoj4lSCQMlOMc8cCWtaKjWCQElqL4GNNatcIPiGT/xpi0+FrqQM6E3LgjIL9QfaphLW2dZKWQMlVatcIOx8doPyCOJSoTMxDXpSzdCXkf0wuoFIaV6PGB6lvCL++SAqk5lA4JCJc9H0Ug8Q8htM/nSsNkF3Sib0pmcRBJ1BpLSYCYN9FMwXXDim5lw47mYB8bkMhD7DJzoSOtem45EgA6A3iOS2TYTAURSlCcx3FxwL31KYntvZkXdCBkKP4RMeCVvWZcqKLgqIpPbSh32D8yhs2VrvEszj667OvFIGEDw/Pk9pXyU7EkQDQU1kcwoV3lkU/2/nXGLbqKIwPHsW3bFgUakLltmxCBJlwaMgKkWqWDUP23EmtmfGM01KSJWWOkqa1LFjOw/aOM82kFYlEQIEgsIirBAPoUqkLCBIIERBpaIsQGrThJLhnihwgImUzLkz4zuj+0tHspSqXZyv57/3Pzc+005NKylnCfLmszxs3h8f63iMCgSGT/yWIDQQieE0NpwCxekWQoMQClhv73GHwbUGH6lYN+nP8PG7G6iWEAog4iULG02Fgv+hDKSOOC2cIMDugv5oBu2iSgMCwyeKJYgPBIZP2GBeKPrbPHsos3hlDDIGmAbw2ZO/t1wy/yiPnXyUCAR+d4MzMyBYgkBAIAyn8BDJD4Xg7yvxduEAgrrBHGkiWIKwQOQc4ZMHUOCUEKZwOsBhkgIEhk/bB8TxeJpgCQIDgeGT59WbO4pXUHFqE88OroDA8AlsYTxBmAaiA4Hhky+lDjXZA30xQUDAZJJNh30UIOD5PFgDToSoAIHhEzbQTyjOqkLAUCmb6wyGOgXkFojppw45pkKkgMDwKRgoipoI54ZGBeQWiIuPH7TPqbhpjCIQiWE1CBgQikIzQFGzycDODU0KyC0QC/WP2BPJFDQgskDQwyd+KHqH2gOGwbqFk4EAxFQsgU2IIBCxchc0p6Z1oqgGcpsYrVif4JmBAMTskSNiwoBACBI+8ZdWiNl9xYyfFtGDtwkCEPNPPyEyDAgEf/gkVHUWWu18UfdkImyDUGUg7McuE4CAc8NkqxpdIDB8Erayhbj9UjFlF4qGu2lQMjcqI9aXDALNMRGoQMw1NAgMAj8Q8VIHoUm1tZOuQhLOGmArjuova3f6J4yVwVlroG/BeljhFwIB06GqpqMJBIZPoa/kcPNG+lziB/NCe6X7TfNJxVshEBefPcTdqIWTOfujpdftn1a/sX+5ccO+d/fu1uevP/7Uvlqdqh0Q+Hw+tNU+GrttzKgfHH9Nf56B8KCC8gcInmvmXGeXvbL8ob2bAJKlgXywQGD4FLpqK7Xc0arJax2XMz0MggNKUHrvYP0qDwzQaDdanl8IDIh4yZCW4FZvH36ODATAQNFb5VH/gMDwKXyWsKjH0BJqpKWjjd9TGvP5O+/aVP1++1f/gMDn82GyhDpFFF2KJX6mNAYOjSShdVCBCGH4hAWWYMyqF8ASWD2giKaZtL7mtikw8nn13Rcr3gOBz+eFKrXS8pvDEkTVVCa7RrYLumDCUIEQPHxCS8icb72+kyVIIHaWN0Dg83lpCRKI/4RP0hJqBQQkkijOmwY/EHiIDNoSJlq/3baEeiUqqmazP7ptCiSOnIJImwSEMXiqpuFTajx+MzunLqIlREwzmnmV0hj4H44ihVN0IPD5fGCWoE8ll19Y1DWHJUQQiCFKY2BhRRQsvajBFCF8osfIYAnWK6mcwxKirDndOlxNG38SmgNjn3LdhM0oCYahzo5/P5+XluATEPvYwfKei8ZQoQCb4dp4dudO/B0+SUvw2TaukJqE9rHrmQJW5LAd5fl31OJpeD7vqSUcezVd2LIEqd1tg3D7gIwCmg/RNHyGnQUfCGgXED55aQkNW5YgtbOmNfMrkZ/OHet/EQ6R5MxAn277bNsSDihSe5oSjTglBCmcDhA+kR6bSEvwIJMQraw+1ZUlsGnwhrQEb6ZEHbtx3BIJht7u1J4toeNSxpKW4Id1ZLL3RYChZGakJYigWd2aqDUUL2u6reabd7QE2CyCJcjMIPJQIAzGmZjDEmCzKC0hrFDQbQJgcD5JlxLnTBHUQXPweMbWKvF/fktJWoK4UOyHK6lf0+K8bqyfHTSuSUsI4bSARNMjMDbZ5FmftMzLuXnzGUUq3LsPWIjBltRlurk5mcluAFTsfNJTylsPKVLRWp0DHKy5ebAUBsgqq7X/FwPgffbzSfZnNVZ1ipSUlJRUiPQXQ5W7HO3VUgwAAAAASUVORK5CYII=",
  "Bindings": [],
  "View": {
    "Dynamo": {
      "ScaleFactor": 1.0,
      "HasRunWithoutCrash": true,
      "IsVisibleInDynamoLibrary": true,
      "Version": "2.11.0.3973",
      "RunType": "Manual",
      "RunPeriod": "1000"
    },
    "Camera": {
      "Name": "Background Preview",
      "EyeX": -17.0,
      "EyeY": 24.0,
      "EyeZ": 50.0,
      "LookX": 12.0,
      "LookY": -13.0,
      "LookZ": -58.0,
      "UpX": 0.0,
      "UpY": 1.0,
      "UpZ": 0.0
    },
    "NodeViews": [],
    "Annotations": [],
    "X": 0.0,
    "Y": 0.0,
    "Zoom": 1.0
  }
}
```

**This pull request does**

- Adds new `Thumbnail` property to the workspace model, which stores a Base64 encoded string. 

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers
@saintentropy 
@nate-peters 

### FYIs
@mjkkirschner

### CODE REVIEW

**Review by nate-peters — APPROVED** [review]
LGTM! @SHKnudsen

**Comment by mjkkirschner** [comment]
@SHKnudsen the tests ran so long ago that the console logs are now gone - I've started a new build here:
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/
please check it out for any failures when it is complete.

**Comment by mjkkirschner** [comment]
only one test failure:
```
MESSAGE:
  Expected: True
  But was:  False

+++++++++++++++++++
STACK TRACE:
   at NUnit.Framework.Assert.That(Object actual, IResolveConstraint expression, String message, Object[] args)
   at DynamoCoreWpfTests.SerializationTests.JSONisSameBeforeAndAfterSaveWithDummyNodes() in c:\WorkspaceDynamo\test\DynamoCoreWpfTests\SerializationTests.cs:line 444
```

**Comment by mjkkirschner** [comment]
@SHKnudsen either is fine with me - but guess it's better to update the test file if it's not too onerous?

File: src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs [hunk 1]
```diff
@@ -203,6 +203,8 @@ internal int CurrentPasteOffset
         private DateTime lastSaved;
         private string author = "None provided";
         private string description;
+        private string thumbnail;
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs:206]: before this gets merged- a quick thought, do all of these new properties make sense on both home workspaces and custom nodes workspaces?

Comment (SHKnudsen) [reply src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs:206]: Tbh I hadn't considered that this also affected the CN workspace. Most of these probably don't make sense on the CN, maybe the `Author` is okay, but not sure i see the need for the others.

@nate-peters @saintentropy, any thoughts?

Comment (SHKnudsen) [reply src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs:206]: Also, just noticed that its the same with the new `ExtensionWorkspaceData` not sure that makes sense in the CN workspace either?

**Review by mjkkirschner — APPROVED** [review]

**Pull request merged**